src/components/MediaCard/MediaCard.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {useState, useCallback} from 'react';
+import {HorizontalDotsMinor} from '@shopify/polaris-icons';
+
+import {classNames} from '../../utilities/css';
+import {useI18n} from '../../utilities/i18n';
+import {Action, ActionListItemDescriptor} from '../../types';
+import {Card} from '../Card';
+import {Button, buttonFrom} from '../Button';
+import {Heading} from '../Heading';
+import {Popover} from '../Popover';
+import {ActionList} from '../ActionList';
+import {ButtonGroup} from '../ButtonGroup';
+import {Stack} from '../Stack';
+
+import styles from './MediaCard.scss';
+
+interface MediaCardProps {
+  /** The visual media to display in the card */
+  children: React.ReactNode;
+  /** Heading content */
+  title: string;
+  /** Body content */
+  description: string;
+  /** Main call to action, rendered as a basic button */
+  primaryAction: Action;
+  /** Secondary call to action, rendered as a plain button */
+  secondaryAction?: Action;
+  /** Action list items to render in ellipsis popover */
+  popoverActions?: ActionListItemDescriptor[];
+  /** Whether or not card content should be laid out vertically
+   * @default false
+   */
+  portrait?: boolean;
+}
+
+export function MediaCard({
+  title,
+  children,
+  primaryAction,
+  secondaryAction,
+  description,
+  popoverActions = [],
+  portrait = false,
+}: MediaCardProps) {
+  const i18n = useI18n();
+  const [popoverActive, setPopoverActive] = useState(false);
```

Comment:
We've got a [`useToggle`]()https://github.com/Shopify/polaris-react/blob/master/src/utilities/use-toggle.ts helper that helps encapsulate the callback wrangling:

```
const {value: popoverActive, toggle: togglePopoverActive} = useToggle(false);